libraries/AP_Compass/AP_Compass.cpp

```diff
@@ -464,6 +464,10 @@ Compass::Compass(void) :
     _null_init_done(false),
     _hil_mode(false)
 {
+    if (_singleton != nullptr) {
+        AP_HAL::panic("Compass must be singleton");
```

Comment:
Damn, I missed this. We should only panic in SITL right? That's what we have been doing in other classes, unless I missed it too.